lib/blemulator/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -1,43 +1,62 @@
 part of internal;
 
 class PlatformToDartBridge {
-
   SimulationManager _manager;
-  MethodChannel _upstreamChannel;
+  MethodChannel _platformToDartChannel;
 
   PlatformToDartBridge(this._manager) {
-    _upstreamChannel = new MethodChannel(ChannelName.upstream);
-    _upstreamChannel.setMethodCallHandler(_handleCall);
+    _platformToDartChannel = new MethodChannel(ChannelName.platformToDart);
+    _platformToDartChannel.setMethodCallHandler(_handleCall);
   }
 
   Future<dynamic> _handleCall(MethodCall call) {
     switch (call.method) {
       case DartMethodName.createClient:
-        return _createClient();
+        return _createClient(call);
       case DartMethodName.destroyClient:
-        return _destroyClient();
+        return _destroyClient(call);
       case DartMethodName.startDeviceScan:
-        return _startDeviceScan();
+        return _startDeviceScan(call);
       case DartMethodName.stopDeviceScan:
-        return _stopDeviceScan();
+        return _stopDeviceScan(call);
+      case DartMethodName.connectToPeripheral:
+        return _connectToDevice(call);
+      case DartMethodName.isPeripheralConnected:
+        return _isDeviceConnected(call);
+      case DartMethodName.disconnectOrCancelConnectionToPeripheral:
+        return _disconnectOrCancelConnection(call);
       default:
-        throw UnimplementedError("${call.method} is not implemented");
+         return Future.error(UnimplementedError("${call.method} is not implemented"));
     }
   }
 
-  Future<void> _createClient() async {
+  Future<void> _createClient(MethodCall call) async {
     await _manager._createClient();
   }
 
-  Future<void> _destroyClient() async {
+  Future<void> _destroyClient(MethodCall call) async {
     await _manager._destroyClient();
   }
 
-  Future<void> _startDeviceScan() async {
+  Future<void> _startDeviceScan(MethodCall call) async {
     await _manager._startDeviceScan();
   }
 
-  Future<void> _stopDeviceScan() async {
+  Future<void> _stopDeviceScan(MethodCall call) async {
     await _manager._stopDeviceScan();
   }
+
+  Future<void> _connectToDevice(MethodCall call) async {
+    await _manager._connectToDevice(call.arguments[ArgumentName.id] as String);
```

Comment:
Why can't we just return the value from manager's connectToDevice method? I think there is no need for making this methods async and performing await here since final await is being done in _handleAsMethodCall in platform_channel.dart